### PR TITLE
troubleshooting: add calico upgrade guide

### PR DESCRIPTION
**Pull request by robszumski**
Adds manifests for the updated RBAC rules and the migration job.

We can link to this from the release notes.

### CODE REVIEW

File: Documentation/troubleshooting/upgrade-calico.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Upgrading Calico Manually
+
+Tectonic included alpha support for Network Security Policy and BGP routing functionality in 1.7.x releases, powered by Calico.
+
+Under the hook, Calico utilized Third Party Resources (TPRs) for configuration. These TPRs need to be upgraded to Custom Resource Definitions (CRDs) before upgrading to 1.8.x. This process is automated, but must be started manually.
```
Comment (bison) [inline Documentation/troubleshooting/upgrade-calico.md]: Under the hood?

Comment (zbwright) [inline Documentation/troubleshooting/upgrade-calico.md]: ... hood ... These TPRs must be upgraded to ...  The process is automated, but must be ,manually started.

File: Documentation/troubleshooting/upgrade-calico.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+After the alpha period, upgrades to Tectonic's network functionality will be fully supported by Tectonic's trademark Automated Operations.
+
```
Comment (zbwright) [inline Documentation/troubleshooting/upgrade-calico.md]: trademarked (is it really?)

File: Documentation/troubleshooting/upgrade-calico.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+## Run the migration Job
+
+An automated migration is run as a Kubernetes Job. [Download the Job object][download-job] and create it in the `default` namespace:
+
+```
+$ kubectl -n default apply -f upgrade-job.yaml
+```
+
+You can check the status of the Job and it's output logs:
```
Comment (zbwright) [inline Documentation/troubleshooting/upgrade-calico.md]: Check the status of the Job and its output logs:

File: Documentation/troubleshooting/upgrade-calico.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+After the job is complete, verify that new data exists as CRDs:
+
+```
+$ kubectl get ippools.crd.projectcalico.org
+$ kubectl get globalfelixconfigs.crd.projectcalico.org
+```
+
+## Upgrade the Role Based Access Control rules
```
Comment (zbwright) [inline Documentation/troubleshooting/upgrade-calico.md]: Role-Based (k8s is hyphenating this)

File: Documentation/troubleshooting/upgrade-calico.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+$ kubectl delete thirdpartyresources/global-bgp-peer.projectcalico.org
+$ kubectl delete thirdpartyresources/global-config.projectcalico.org
+$ kubectl delete thirdpartyresources/ip-pool.projectcalico.org
+$ kubectl delete thirdpartyresources/system-network-policy.alpha.projectcalico.org
+$ kubectl delete -f upgrade-job.yaml
+```
+
+## Proceed with your Tectonic upgrade
+
+This process ensures that TPRs utilized by Calcio are removed. If you are using other TPRs, remove or migrate them before attempting to upgrade to Tectonic 1.8.
```
Comment (zbwright) [inline Documentation/troubleshooting/upgrade-calico.md]: Calico

File: Documentation/troubleshooting/upgrade-calico.md (hunk 2 above)
Comment (robszumski) [reply Documentation/troubleshooting/upgrade-calico.md]: more of a turn of phrase

Comment (diegs) [reply Documentation/troubleshooting/upgrade-calico.md]: Under the hood?

**Pull request merged**